Dockerfile

```diff
@@ -0,0 +1,34 @@
+FROM rust:1.67-bullseye AS chef
+
+RUN cargo install cargo-chef; \
+    rustup component add rustfmt;
+WORKDIR /app
+
+FROM chef AS planner
+COPY . .
+RUN cargo chef prepare --recipe-path recipe.json
+
+FROM chef AS builder
+
+COPY --from=planner /app/recipe.json recipe.json
+RUN cargo chef cook --release --recipe-path recipe.json
+COPY . .
+# RUN cargo build --release --bin chainsaw-demo-grpc
+RUN cargo build --release
+
+FROM debian:bullseye-slim AS base-runtime
+# Running apt again here to install common certificate authorites on the
+# base-runtime target.
+#
+# The previous install including `cmake` is dropped as this layer is independent
+# of the previous.
+RUN apt-get update
+RUN apt-get install ca-certificates -y
+
+WORKDIR /app
+COPY cactuar.toml cactuar.toml
+
+FROM base-runtime AS cactuar
+COPY --from=builder --chown=root:root /app/target/release/controller /usr/local/bin/
+
+CMD ["controller"]
```

Comment:
I'd recommend replacing this with the distroless container e.g. https://github.com/GoogleContainerTools/distroless/blob/main/examples/rust/Dockerfile

it does contain ca-certs AFAIK - although now that I think about it why would we need them?

Reply:
This is another thing ripped from Chainsaw, agreed on `ca-certs`, I'll take a look at distroless as well.